### PR TITLE
V2 aatish

### PR DESCRIPTION
**Pull request by aatish17varma**
Worked on Job Page

### CODE REVIEW

**Comment by aatish17varma** [comment]
Added Expire Date

**Comment by ashleytqy** [comment]
@aatish17varma 
- change base branch from: `TechAtNYU:ash.v2-cal` to `TechAtNYU:v2`
- good practice for future PRs: add images when useful (i.e. in this case)
- React style guide: https://github.com/airbnb/javascript/tree/master/react

## To-do
- [ ] job link opens a new page of that link
- [x] `apply` button on job card to be red
- [ ] format for job card: "Digital Product Intern / Company / Apply by: [expiry date in human readable format]
- [ ] render html in job modal & make it look pretty

screen shot 2017-10-25 at 12 17 16 am
screen shot 2017-10-25 at 12 17 20 am

File: src/Jobs.js [hunk 1]
```diff
@@ -4,27 +4,63 @@
 
 import React, { Component } from 'react'
 import Constants from './Constants';
+import {IndexLink} from "react-router";
```
Comment (ashleytqy) [inline src/Jobs.js]: minor: let's stick to having spaces when we deconstruct `{ IndexLink }`

File: src/Jobs.js [hunk 2]
```diff
@@ -4,27 +4,63 @@
...
-                    <span>{this.props.job.paid} | {this.props.job.salary} </span><br/><br/>
-                    <div>{this.props.job.description}</div>
+                    <span className="job-title">{this.props.job.attributes.positionTitle}</span><br/>
+                    <span> {this.props.job.attributes.positionLevel} </span><br/><br/>
+                    <div>Expires At: {this.props.job.attributes.exiresAt}</div>
                 </div>
                 <div className="job-apply">
-                    <p> APPLY </p>
-                </div>
+                    {<RaisedButton onClick = {() => {this.setState({modalIsOpen: true})}} label = "Apply" primary = {true} fullWidth = {true} />}
```
Comment (ashleytqy) [inline src/Jobs.js]: style: separate the attributes into new lines for readability and no spaces before & after `=`
```
<RaisedButton
  onClick={() => {this.setState({modalIsOpen: true})}}
  label="Apply"
  primary={true}
  fullWidth={true} />
```

File: src/Jobs.js [hunk 3]
```diff
@@ -4,27 +4,63 @@
...
+                </div>       
+        <Modal
```
Comment (ashleytqy) [inline src/Jobs.js]: indentation

File: src/Jobs.js [hunk 4]
```diff
@@ -4,27 +4,63 @@
...
 };
 
-
+const jobsURL = 'https://api.tnyu.org/v3/jobs'
 class JobsCard extends Component {
     constructor(props) {
         super(props);
+        this.state =  {
+            modalIsOpen: false,
+            customStyles: {
```
Comment (ashleytqy) [inline src/Jobs.js]: css should be in a separate file

File: src/Jobs.js [hunk 5]
```diff
@@ -4,27 +4,63 @@
 
 import React, { Component } from 'react'
 import Constants from './Constants';
+import {IndexLink} from "react-router";
+import Modal from "react-modal"
+import axios from "axios"
```
Comment (ashleytqy) [inline src/Jobs.js]: don't think you need to use `axios`, native `fetch` would do just fine

File: src/Jobs.js [hunk 6]
```diff
@@ -4,27 +4,63 @@
...
+import RaisedButton from 'material-ui/RaisedButton'
 
 const colorStyle = {
     color: "#f05158"
 };
 
-
+const jobsURL = 'https://api.tnyu.org/v3/jobs'
```
Comment (brentdur) [inline src/Jobs.js:16]: Can load from `v3-test` if that helps with testing

File: src/Jobs.js [hunk 7]
```diff
@@ -80,9 +132,10 @@ class Jobs extends Component{
                     </div>
                 </div>
                 <div id="jobs-container">
-                    {placeholder.map( (obj, index) => {
+                    {console.log('In render method', this.state.data)}
+                    {this.state.data.length !== 0 ? this.state.data.map( (obj, index) => {
                         return <JobsCard key={index} job={obj}/>
-                    })}
+                    }) : console.log("")}
```
Comment (brentdur) [inline src/Jobs.js]: `console.log`?

**Comment by brentdur** [comment]
Please don't commit `package-lock.json`. Add it to `.gitignore`

**Comment by aatish17varma** [comment]
screen shot 2018-02-03 at 7 37 41 pm
screen shot 2018-02-03 at 7 37 29 pm

Gotta work on:
- Adding CSS to the buttons in the modal

**Comment by ashleytqy** [comment]
You should also change the title to something more specific, like – Jobs Page, and in the description, add the url to the issue you are fixing. @aatish17varma

File: src/main.js [hunk 8]
```diff
@@ -4,9 +4,18 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import App from './App';
 
+import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
+
+const Apps = () => (
+<MuiThemeProvider>
```
Comment (ashleytqy) [inline src/main.js]: can you remove this?

Comment (ashleytqy) [inline src/main.js:10]: can you remove `MuiThemeProvider`?

File: stylesheets/index.scss [hunk 9]
```diff
@@ -333,6 +333,23 @@ hr {
...
+    margin: auto;
+    padding: 0;
+    border: 1px solid #888;
+    width: 80%;
+    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2),0 6px 20px 0 rgba(0,0,0,0.19);
+    animation-name: animatetop;
+    animation-duration: 0.4s
+}
+
+
```
Comment (ashleytqy) [inline stylesheets/index.scss:350]: extra lines

File: src/Jobs.js [hunk 10]
```diff
@@ -2,68 +2,104 @@
...
+            data: []
+        }
+    }
+     componentDidMount(){
+         fetch(jobsURL, {
+            method:'GET',
+        })
+            .then((response) => response.json())
+            .then((responseData) => {
+                console.log('responseData', responseData)
```
Comment (ashleytqy) [inline src/Jobs.js:99]: remove `console.log`s

File: src/Jobs.js [hunk 11]
```diff
@@ -2,68 +2,104 @@
...
+                    <div>Expires At: {this.props.job.attributes.exiresAt.substring(0, this.props.job.attributes.exiresAt.indexOf('T'))}</div>
+                    <button className = "close-modal" onClick = {() => {this.setState({modalIsOpen: false})}}> Close Modal</button>
+                  </div>
                 </div>
+                :
+                console.log("Modal is currently closed")
+              }
             </div>
-        )
+              )
```
Comment (ashleytqy) [inline src/Jobs.js:58]: remove commented code

File: src/Jobs.js [hunk 12]
```diff
@@ -2,68 +2,104 @@
...
+                  <div className="job-content" >
+                      <span className="job-title">{this.props.job.attributes.positionTitle}</span><br/>
+                      <span> {this.props.job.attributes.positionLevel} </span><br/><br/>
+                      <div>Expires At: {this.props.job.attributes.exiresAt.substring(0, this.props.job.attributes.exiresAt.indexOf('T'))}</div>
+                  </div>
+                  <div className="job-apply">
+                      <p> <button className = "job-apply" onClick ={() => {this.openModal()}}> Apply </button> </p>
+                  </div>
+              </div>
+              { this.state.modalIsOpen == true ?
```
Comment (ashleytqy) [inline src/Jobs.js:44]: `this.state.modalIsOpen == true` => `this.state.modalIsOpen`

File: package.json [hunk 13]
```diff
@@ -11,6 +11,7 @@
   "author": "Adisa Narula",
   "license": "ISC",
   "dependencies": {
+    "axios": "^0.17.0",
```
Comment (ashleytqy) [inline package.json:14]: remove unused package

File: package.json [hunk 14]
```diff
@@ -20,13 +21,16 @@
     "extract-text-webpack-plugin": "^1.0.1",
     "file-loader": "^0.10.1",
     "json-loader": "^0.5.4",
+    "material-ui": "^0.19.4",
```
Comment (ashleytqy) [inline package.json:24]: remove unused package

File: package.json [hunk 15]
```diff
@@ -20,13 +21,16 @@
...
     "node-bourbon": "^4.2.8",
     "node-sass": "^4.5.0",
     "react": "^15.4.2",
     "react-addons-css-transition-group": "^15.4.2",
     "react-dom": "^15.4.2",
     "react-media": "^1.5.1",
+    "react-modal": "^3.0.4",
```
Comment (ashleytqy) [inline package.json:31]: remove unused package

File: package.json [hunk 16]
```diff
@@ -20,13 +21,16 @@
...
     "react-router": "^3.0.1",
+    "react-router-dom": "^4.2.2",
```
Comment (ashleytqy) [inline package.json:33]: remove unused package

File: src/Jobs.js [hunk 17]
```diff
@@ -2,68 +2,104 @@
  * Created by adisanarula on 4/7/17.
  */
 
-import React, { Component } from 'react'
+import React, { Component } from 'react';
+import { Router, Route, Switch, Link } from 'react-router';
 import Constants from './Constants';
+import {IndexLink} from "react-router";
+import Modal from "react-modal"
```
Comment (ashleytqy) [inline src/Jobs.js:9]: remove unused package

**Review by ashleytqy — CHANGES_REQUESTED** [review]

**Pull request merged**